PocketKit/Sources/PocketKit/PocketAppDelegate.swift

```diff
@@ -80,6 +80,23 @@ public class PocketAppDelegate: UIResponder, UIApplicationDelegate {
         Textiles.initialize()
         refreshCoordinator.initialize()
 
+        do {
+            let migration = LegacyUserMigration(
+                userDefaults: userDefaults,
+                encryptedStore: PocketEncryptedStore(),
+                appSession: appSession
+            )
+
+            let attempted = try migration.perform()
+            if attempted {
+                Crashlogger.breadcrumb(category: "launch", level: .info, message: "Legacy user migration required; running.")
+            } else {
+                Crashlogger.breadcrumb(category: "launch", level: .info, message: "Legacy user migration not required; skipped.")
+            }
+        } catch {
+            Crashlogger.capture(error: error)
+        }
```

Comment:
can we bypass performing the migration early on by checking if `version was < 8.x.x and the migration has not been run here`?

Reply:
The (current) problem is `version` is read at decryption-time, which is when `perform` is called. I think this can be refactored, though.